File: TransCoder_tokenized_test_set_functions/java/LARGEST_INCREASING_SUBSEQUENCE_OF_CONSECUTIVE_INTEGERS.java
static void findLIS ( int A [ ], int n ) {
  Map < Integer, Integer > hash = new HashMap < Integer, Integer > ( );
  int LIS_size = 1;
  int LIS_index = 0;
  hash . put ( A [ 0 ], 1 );
  for ( int i = 1;
  i < n;
  i ++ ) {
    hash . put ( A [ i ], hash . get ( A [ i ] - 1 ) == null ? 1 : hash . get ( A [ i ] - 1 ) + 1 );
    if ( LIS_size < hash . get ( A [ i ] ) ) {
      LIS_size = hash . get ( A [ i ] );
      LIS_index = A [ i ];
    }
  }
  System . out . println ( "LIS_size = " + LIS_size );
  System . out . print ( "LIS : " );
  int start = LIS_index - LIS_size + 1;
  while ( start <= LIS_index ) {
    System . out . print ( start + " " );
    start ++;
  }
}

File: TransCoder_tokenized_test_set_functions/python/LARGEST_INCREASING_SUBSEQUENCE_OF_CONSECUTIVE_INTEGERS.py
def findLIS ( A, n ) :
    hash = dict ( )
    LIS_size, LIS_index = 1, 0
    hash [ A [ 0 ] ] = 1
    for i in range ( 1, n ) :
        if A [ i ] - 1 not in hash :
            hash [ A [ i ] - 1 ] = 0
        hash [ A [ i ] ] = hash [ A [ i ] - 1 ] + 1
        if LIS_size < hash [ A [ i ] ] :
            LIS_size = hash [ A [ i ] ]
            LIS_index = A [ i ]
    print ( "LIS_size =", LIS_size )
    print ( "LIS : ", end = "" )
    start = LIS_index - LIS_size + 1
    while start <= LIS_index :
        print ( start, end = " " )
        start += 1
    
